include/xgboost/tree_model.h

```diff
@@ -388,67 +388,67 @@ class TreeModel {
       }
       nodes.push_back(node_values);
     }
-    (*p_json)["nodes"] = json::Array(nodes);
-    std::vector<json::Json> leaf_arr;
+    (*p_kvstore)["nodes"] = serializer::Array(nodes);
+    std::vector<serializer::NestedKVStore> leaf_arr;
     for (bst_float leaf : leaf_vector_) {
-      leaf_arr.push_back(json::Number{leaf});
+      leaf_arr.push_back(serializer::Number{leaf});
     }
-    (*p_json)["leaf_vector"] = json::Array(leaf_arr);
+    (*p_kvstore)["leaf_vector"] = serializer::Array(leaf_arr);
   }
 
-  virtual void Load(json::Json* p_json) {
-    auto& r_json = *p_json;
-    json::InitParametersFromJson(r_json, "TreeParam", &param);
+  virtual void Load(serializer::NestedKVStore* p_kvstore) {
+    auto& r_kvstore = *p_kvstore;
+    serializer::InitParametersFromKVStore(r_kvstore, "TreeParam", &param);
 
-    std::vector<json::Json> const& nodes_json =
-        json::Get<json::Array>(r_json["nodes"]).GetArray();
+    std::vector<serializer::NestedKVStore> const& nodes_kvstore =
+        serializer::Get<serializer::Array>(r_kvstore["nodes"]).GetArray();
 
-    size_t n_nodes = nodes_json.size();
+    size_t n_nodes = nodes_kvstore.size();
     nodes_.resize(n_nodes);
     stats_.resize(n_nodes);
 
     param.num_nodes = n_nodes;
 
     int nid = 0;
-    for (auto const& node_j : nodes_json) {
-      std::map<std::string, json::Json> const& node_map =
-          json::Get<json::Object const>(node_j).GetObject();
-      nid = json::Get<json::Number const>(node_map.at("nodeid")).GetInteger();
+    for (auto const& node_kv : nodes_kvstore) {
+      std::map<std::string, serializer::NestedKVStore> const& node_map =
+          serializer::Get<serializer::Object const>(node_kv).GetObject();
+      nid = serializer::Get<serializer::Integer const>(node_map.at("nodeid")).GetInteger();
       auto& node = nodes_.at(nid);
 
       if (node_map.find("leaf") != node_map.end()) {
         node.cleft_ = -1;
         node.SetLeaf(
-            json::Get<json::Number const>(node_map.at("leaf")).GetFloat());
```

Comment:
@trivialfis I'll add `GetFloat()` method to `NumberValue`. BTW, I added `IntegerValue` to prevent integer-to-float conversion, which can be catastrophic.